aten/src/THCUNN/SpatialClassNLLCriterion.cu

```diff
@@ -123,8 +123,9 @@ __global__ void cunn_SpatialClassNLLCriterion_sizeAverage_kernel(
           T *output,
           T *total_weight)
 {
-  if (*total_weight > 0)
+  if (*total_weight >= 0) {
```

Comment:
Yeah, I feel like the original condition, `*total_weight > 0`, was used to guard against division by zero and we might not have to worry about negative numbers.

I'm a little worried about BC -- I'm not sure if any users relied on the total_weight = 0 behavior

Reply:
So, concrete action items:
(1) we should try to find some example of this type of reduction, dividing by weights, in another framework and see what happens
(2) we should think about a use case where total_weight = 0 and use that to inform our decision of if we want to return 0 or NaN

Reply:
I investigated this. Let me rephrase my findings.